Migrate from san to vue

jetfuel opened the pull request:
We created Vue version of the VisualDL. Please take some time to take a look. Once it is merged, let's discuss how we can build interactive graph together.

jetfuel committed:
Initial checking to setup Vue basic requirements. (#264)

Add App.vue
Add AppMenu.vue

jetfuel committed:
Vue with app menu (#265)

* Add vuetify framework
Update AppMenu.vue to include the basic menu item.

* use name as key.

jetfuel committed:
Add major components (#266)

* Add vuetify framework
Update AppMenu.vue to include the basic menu item.

* use name as key.

* Add Scalars, Images, Histograms, Graph.

* Switch to use the stylus language for CSS

* Port functions from Scalars.san to Scalars.vue

jetfuel committed:
Add config vue (#267)

* Add vuetify framework
Update AppMenu.vue to include the basic menu item.

* use name as key.

* Add Config Vue

nickyfantasy committed:
Integrate chart to scalar with Vue (#268)

* Fix warnings from Vue compiler
* use default prop instead of duplicate data and prop
* use function instead of shorthand declaration with data

* refract naming style, fix scalar options

* Fix config UI

* Add chart and chart page and integrate to scalar

* update with varun and jeff's comments

jetfuel committed:
Add images config (#270)

* Add vuetify framework
Update AppMenu.vue to include the basic menu item.

* use name as key.

* Add the Config.vue for Images

* Add the CharPage.vue and Image.vue

* Hook up the config to the image.vue so that the image height and width can update

nickyfantasy committed:
Show Scalar Data and add ExpandPanel (#272)

* Watch tagInfo changes to update chart and fix tooltip issue

* Add expand panel and maintain isShow state in the panel

* Add Expand Panel in Image

jetfuel committed:
Add watch group name reg (#273)

* Add vuetify framework
Update AppMenu.vue to include the basic menu item.

* use name as key.

* Add Watch on the groupNameReg

jetfuel committed:
Allow the nav bar to persist the selected item style. (#274)

jetfuel committed:
Fix the incorrect pagination issue. (#276)

jetfuel committed:
Add the Graph.vue and the Config.vue for Graph tab (#277)

jetfuel committed:
Add the Histogram related vue files. (#278)

jetfuel committed:
Fix error message on charts (#279)

* Make sure to clear the chart before each use.

* Add key for v-for loop.

nickyfantasy committed:
Add theme and fix UI (#280)

-Apply theme color in Vuetify
-Add main.styl to override Vuetify default stylus variables
-Add relative path in web pack config to find variables.styl

nickyfantasy committed:
Fix multiple scalar issues: (#283)

* Fix multiple scalar issues:
-download link / outliers should take boolean value instead of array
-download link now show proper selector of runsItem and download button
-Use tag as key to prevent chart re-rendering
-Add missing label in scalar config and fix UI

* simply use index for key because scalar has different tag info structure from image and histogram

jetfuel committed:
Merge branch 'develop' into migrateFromSanToVue

nickyfantasy committed:
Fix font size and reorganize css and stylus (#287)

daming-lu committed:
Merge branch 'develop' into migrateFromSanToVue

daming-lu commented on frontend/src/common/component/ExpandPanel.vue:
```diff
@@ -0,0 +1,73 @@
+    },
+    methods: {
+        handleHeadClick() {
+            this.toogleShow();
```
toggle

daming-lu commented on frontend/src/common/component/ExpandPanel.vue:
```diff
@@ -0,0 +1,73 @@
+    <div class="visual-dl-expand-panel">
+        <h3
+            class="visual-dl-expand-head"
+            @click="handleHeadClick()"
```
Since this click only toggles show/hide, we can write it simply as below:
```
@click="isShow = !isShow"
```

daming-lu commented on frontend/src/common/component/Notification/index.js:
```diff
@@ -1,7 +1,9 @@
-import Notification from './Notification';
-import NotificationItem from './NotificationItem';
+//TODO: Migrate this part to Vue
```
Can this file be moved?

daming-lu commented on frontend/src/graph/ui/Chart.vue:
```diff
@@ -0,0 +1,195 @@
+
+        getOriginChartsData() {
+            getPluginGraphsGraph().then(({status, data}) => {
+                if (+status === 0 && data.url) {
```
Is the '+' an error?
Also we should add handler for error promise later

daming-lu commented on frontend/src/graph/ui/Config.vue:
```diff
@@ -0,0 +1,61 @@
+
+export default {
+    props:['config'],
+    data() {
```
data() is not used thus should be removed.

daming-lu commented on frontend/src/index.js:
```diff
@@ -1,16 +1,46 @@
-import 'normalize.css/normalize.css';
-import 'san-mui/index.css';
-import './common/component/ui-common.styl';
+//import 'normalize.css/normalize.css';
```
Should we also delete this file?

daming-lu committed:
Merge branch 'develop' into migrateFromSanToVue

Commit:
Fix comments

daming-lu commented:
<img width="1677" alt="screen shot 2018-03-02 at 5 16 37 pm" src="https://user-images.githubusercontent.com/5635322/36892737-a8f4c4b4-1e40-11e8-949a-03996a0a5e13.png">

The download button toggled by 'Show data download link' is broken. Note the small download button (4th one on the upper row) is working

Commit:
placeholder index.js

nickyfantasy commented:
@daming-lu good catch regarding the download button bug. This is because "train" run is empty and nothing can be downloaded, same issue before the migration. I think the fix should let the selector only displays non empty chart.

nickyfantasy committed:
Fix issue where download data link points to empty run (#294)

-Dropdown should only display non empty runs
-Select first run by default
-Remove drop down if all are empty runs
-Refract download type to runItemForDownload
-Add train run and multiple tags in mock data

daming-lu committed:
Merge branch 'develop' into migrateFromSanToVue

daming-lu committed:
Merge branch 'develop' into migrateFromSanToVue

nickyfantasy committed:
Fix error when scalar data empty (#298)

-Error comes in when trying to access elements inside empty data
-When one of the runs has empty data, the chart will display nothing even other run has non empty data
-Fix by checking length of data before accessing

deqingli commented on frontend/src/common/component/AppMenu.vue:
```diff
@@ -0,0 +1,85 @@
+    },
+    methods: {
+            handleItemClick: function (item) {
+            this.selected = item.name
```
这里为何没有缩进，建议合入后改一下，小问题

deqingli commented on frontend/src/common/component/Notification/index.js:
```diff
@@ -1,7 +1,3 @@
-};
+/**
+ * Created by ludaming on 3/4/18.
+ */
```
这个文件内容为空，是不是可以删了

deqingli commented on frontend/src/graph/ui/Config.vue:
```diff
@@ -0,0 +1,57 @@
+            Download image
+        </v-btn>
+
+        <v-slider label="Scale"
```
这个 label 属性可以写在下一行，格式统一

deqingli commented on frontend/src/images/ui/ChartPage.vue:
```diff
@@ -0,0 +1,89 @@
+<script>
+import ExpandPanel from '../../common/component/ExpandPanel';
+import Image from './Image';
+//import Pagination from 'san-mui/Pagination';
```
san-mui 相关的后面统一删了吧

deqingli reviewed (approved):
LGTM

jetfuel committed:
Merge branch 'develop' into migrateFromSanToVue
